src/open_inwoner/kvk/views.py

```
@@ -90,6 +91,34 @@ def post(self, request):
             # Directly calling `super().form_invalid(form)` would override the error
             return self.render_to_response(context)
 
-        request.session[KVK_BRANCH_SESSION_VARIABLE] = request.POST["branch_number"]
+        branch_number = request.POST["branch_number"]
+        request.session[KVK_BRANCH_SESSION_VARIABLE] = branch_number
+
+        # data for update company signal
+        company_branch = (
+            next(
+                branch
+                for branch in form.company_branches
+                if branch.get("vestigingsnummer", None) == branch_number
+            )
+            or form.company_branches[0]
+        )
```

Should raise if `branch_number` is not found, with a meaningful error message (of the Not authorized variety).

---

The form raises a ValidationError if the branch number from the POST doesn't correspond to any branch number returned by the KVK API